WcaOnRails/app/models/user.rb

```diff
@@ -402,10 +402,18 @@ def any_kind_of_delegate?
     delegate_status.present?
   end
 
+  def is_a_senior_delegate?
```

Comment:
Call me an OCD person, but I'd prefer this to be simply "senior_delegate?" since that's what we use on all the rest (IIRC).

Reply:
Yeah, thanks for commenting on this. The problem is that we currently have a `senior_delegate` method that returns a users senior delegate (if they have one). With your proposal, we'd have two methods:

1. `def senior_delegate?` - Is this user a senior delegate?
2. `def senior_delegate` - Give me this users's senior delegate, if they have one.

I think that's too confusing. Maybe we could rename them as follows?

1. `def senior_delegate?` - Is this user a senior delegate?
2. `def get_senior_delegate` or even `def get_my_senior_delegate`? - Give me this users's senior delegate, if they have one.

What do you think is best, @pedrosino? (CC-ing @jonatanklosko, @viroulep, and @larspetrus just in case they have any clever naming ideas as well.)

Reply:
Just some general thoughts:
I don't think adding `senior_delegate?` changes the way you think of `senior_delegate`. If the former is ambiguous whether it is a user or a boolean value then changing it would've made sense at any point. Given that we've been living with this so long and no one complained IIRC, it seems clear enough.
At least I'm quite opposed to `get_` given how it's defined https://github.com/thewca/worldcubeassociation.org/blob/c9366f9ba7cdb037e661fc7ef335d6b6ded37e0c/WcaOnRails/app/models/user.rb#L73
Achieving the `get_` behaviour would either mean having `belongs_to :get_senior_delegate`, which looks terribly or setting up an alias, which would result in some people using the alias and some using `senior_delegate` directly, thus increasing the complexity. I might have missed some cleaner way, but these two came to my mind.

Reply:
Ah, in this case I think it's fine to keep is_a_senior_delegate.
Or change the current senior_delegate to my_senior_delegate

Reply:
@jonatanklosko, I was thinking we would rename the method as per the instructions in https://coderwall.com/p/gutg6g/rename-a-method-in-ruby, but the fact that you're okay with it means I'm okay with it. I'll implement a `def senior_delegate` and a `def senior_delegate?`